tests/sphinx/test_base.py

```diff
@@ -253,27 +268,25 @@ def test_set_kpoints(self):
         center_shift = [0.1, 0.1, 0.1]
 
         trace = {"my_path": [("GAMMA", "H"), ("H", "N"), ("P", "H")]}
-        kpoints_dict = odict([('kPoints',
-                               odict([('relative', None),
-                                      ('from',
-                                       odict([('coords', '[0.0, 0.0, 0.0]'),
-                                              ('label', '"GAMMA"')])),
-                                      ('to___0',
-                                       odict([('coords', '[0.5, -0.5, 0.5]'),
-                                              ('nPoints', 20),
-                                              ('label', '"H"')])),
-                                      ('to___1',
-                                       odict([('coords', '[0.0, 0.0, 0.5]'),
-                                              ('nPoints', 20),
-                                              ('label', '"N"')])),
-                                      ('to___1___1',
-                                       odict([('coords', '[0.25, 0.25, 0.25]'),
-                                              ('nPoints', 0),
-                                              ('label', '"P"')])),
-                                      ('to___2',
-                                       odict([('coords', '[0.5, -0.5, 0.5]'),
-                                              ('nPoints', 20),
-                                              ('label', '"H"')]))]))])
+
+        kpoints_group = {
+            'relative': True,
+            'from': odict([
+                ('coords', '[0.0, 0.0, 0.0]'), ('label', '"GAMMA"')
+            ]),
+            'to___0': odict([
+                ('coords', '[0.5, -0.5, 0.5]'), ('nPoints', 20), ('label', '"H"')
+            ]),
+            'to___1': odict([
+                ('coords', '[0.0, 0.0, 0.5]'), ('nPoints', 20), ('label', '"N"')
+            ]),
+            'to___1___1': odict([
+                ('coords', '[0.25, 0.25, 0.25]'), ('nPoints', 0), ('label', '"P"')
+            ]),
+            'to___2': odict([
+                ('coords', '[0.5, -0.5, 0.5]'), ('nPoints', 20), ('label', '"H"')
+            ]),
```

Comment:
This is ugly.
This could be nicer:
```
'to[]' = [odict([('coords', '[0.5, -0.5, 0.5]'), ('nPoints', 20), ('label', '"H"')]),
          odict([('coords', '[0.0, 0.0, 0.5]'), ('nPoints', 20), ('label', '"N"')]),
          odict([('coords', '[0.25, 0.25, 0.25]'), ('nPoints', 0), ('label', '"P"')]),
          odict([('coords', '[0.5, -0.5, 0.5]'), ('nPoints', 20), ('label', '"H"')])]
```
Technically, it might even be easy to omit the ```[]``` marker in the identifier name ```to[]```; but it might be easier for the user to identify this group that appears multiple times.

It also might be convenient to have a auxiliary function to construct such a list of odicts from a pandas table (using column name for the parameter name).

Reply:
We started to use https://github.com/psf/black - an automated code formatter for Python, which caused some of the line breaks. In addition I would like to emphasise that we follow the python coding conventions - PEP8 - as much as possible to make it easy for Python developers to understand and modify the code later on.

Reply:
Sam is too clever for me - the list-of-dicts is already implemented. But we agreed that the triple-underscore notation also should be kept for the time being, so we probably leave the test as is (maybe add a comment how the list notation would look like, and that the present version tests the underscore version?)